src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -67,17 +78,65 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         steps = FormHelper.separateByNewlines(stepsResponse);
       }
 
+      // recipe is private and stays private
       recipeEntity.setProperty("name", name);
       recipeEntity.setProperty("tags", tags);
       recipeEntity.setProperty("description", description);
       recipeEntity.setProperty("ingredients", ingredients);
       recipeEntity.setProperty("steps", steps);
-      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+      recipeEntity.setProperty("imageBlobKey", imageBlobKey);
+
+      if (privacy.equals("public")) {
+        recipeEntity.setProperty("published", true);
+      }
+      if (privacy.equals("private")) {
+        recipeEntity.setProperty("published", false);
+      }
+
+      // recipe was private and is made public
+      if (!published && privacy.equals("public")) {
+        Entity publicRecipeEntity = FormHelper.copyToNewPublicRecipe(recipeEntity);
+        datastore.put(publicRecipeEntity);
+      }
+
+      // recipe was public
+      if (published) {
+        // make filters to the public recipe
+        Filter nameFilter = new FilterPredicate("name", FilterOperator.EQUAL, savedName);
+        Filter descriptionFilter = new FilterPredicate("description", FilterOperator.EQUAL, savedDescription);
+        // TODO: further filter by author ID
+        Filter composFilter = CompositeFilterOperator.and(nameFilter, descriptionFilter);
```

Comment:
same comment here as above, about either mapping private and public ideas, or enforcing this assumption when a user creates a recipe

Reply:
will do!